raspberry/checkout/checkout_main.py

```diff
@@ -62,16 +65,40 @@ def process_rfid_card(self, uid):
             print("Item already scanned")
             self.interactions.buzz()
             self.interactions.indicate_error()
+
+    def cancel_action(self):
+        if self.state == 1:
+            return self.cancel_checkout
+        else:
+            return self.logic.remove_current_tag
```

Comment:
you should not return anything here. just please, execute the action (this goes for both states). Oh, and feedback would be nice, like displaying REMOVED! when the item is removed